src/components/dialog-manager.js

```diff
@@ -73,15 +73,23 @@ export class DialogManager {
    * @return {!Promise}
    */
   openView(view, hidden = false) {
-    view.whenComplete().catch((reason) => {
+    this.handleView(view);
+    return this.openDialog(hidden).then((dialog) => {
```

Comment:
```suggestion
    return this.openDialog(hidden).then((dialog) => dialog.openView(view));
```

This could be simplified a bit

Reply:
Done.